mlir/lib/Dialect/Rock/Transforms/AlignTiling.cpp

```diff
@@ -628,8 +629,8 @@ makeExtraInputTile(LinalgAlignRewriter &b, TiledOp tiledOp, Value src,
   // block of tiledOp, as their arguments may depend on that control flow. Later
   // we will also move linalg.generic into the block.
   if (laGeneric->getBlock() != tiledOp->getBlock() &&
-      laGeneric.getOperation() != sink)
-    b.setInsertionPoint(sink);
+      isa<ThreadwiseReadIntoOp>(tiledOp))
```

Comment:
I can't help but feel like the logic we're actually looking here involves actually computing the CFG or dominator tree and asking "is the generic in a block that's outside of the one the tiled op is in" or a similar question.

That being said, this here's probably fine.